pkg/scheduler/algorithm/priorities/selector_spreading.go

```diff
@@ -250,43 +235,60 @@ func (s *ServiceAntiAffinity) CalculateAntiAffinityPriorityMap(pod *v1.Pod, meta
 // CalculateAntiAffinityPriorityReduce computes each node score with the same value for a particular label.
 // The label to be considered is provided to the struct (ServiceAntiAffinity).
 func (s *ServiceAntiAffinity) CalculateAntiAffinityPriorityReduce(pod *v1.Pod, meta interface{}, sharedLister schedulerlisters.SharedLister, result framework.NodeScoreList) error {
+	reduceResult := make([]float64, len(result))
+	for _, label := range s.labels {
+		if err := s.updateNodeScoresForLabel(sharedLister, result, reduceResult, label); err != nil {
+			return err
+		}
+	}
+
+	// Update the result after all labels have been evaluated.
+	for i, nodeScore := range reduceResult {
+		result[i].Score = int64(nodeScore)
+	}
+	return nil
+}
+
+// updateNodeScoresForLabel updates the node scores for a single label. Note it does not update the
+// original result from the map phase directly, but instead updates the reduceResult, which is used
+// to update the original result finally. This makes sure that each call to updateNodeScoresForLabel
+// receives the same mapResult to work with.
+func (s *ServiceAntiAffinity) updateNodeScoresForLabel(sharedLister schedulerlisters.SharedLister, mapResult framework.NodeScoreList, reduceResult []float64, label string) error {
 	var numServicePods int64
-	var label string
+	var labelValue string
 	podCounts := map[string]int64{}
 	labelNodesStatus := map[string]string{}
 	maxPriorityFloat64 := float64(framework.MaxNodeScore)
 
-	for _, hostPriority := range result {
+	for _, hostPriority := range mapResult {
```

Comment:
s/host/node everywhere in the file (host is the old name of nodes).

Reply:
let's not do this since this file is going away soon?